bin/pub_insights.dart

```diff
@@ -5,19 +5,64 @@ import 'dart:io';
 import 'dart:typed_data';
 
 import 'package:archive/archive_io.dart' as archive;
+import 'package:args/args.dart';
 import 'package:console_bars/console_bars.dart';
 import 'package:http/http.dart' as http;
 import 'package:path/path.dart' as path;
 import 'package:pub_insights/isolate_worker.dart';
 import 'package:pub_insights/pub_client.dart' as client;
 
+const String _directoryFlag = 'directory';
+const String _helpFlag = 'help';
+const String _fetchModeFlag = 'fetch-mode';
+const String _fetchModeAll = 'all';
+const String _fetchModePopular = 'popular';
+const String _fetchModeFile = 'file';
+const String _fetchModeNone = 'none';
+const String _fetchFileFlag = 'fetch-file';
+const String _optimizeTablesFlag = 'optimize-tables';
+
 Future<void> main(List<String> arguments) async {
-  if (arguments.length != 1) {
-    print('Usage: pub_insights <output path>');
-    return;
+  final ArgParser parser = ArgParser()
+    ..addOption(_directoryFlag,
+        defaultsTo: 'data', abbr: 'd', help: 'Directory to download data into')
```

Comment:
Could we use tall mode for style? Sorry, style wars never end 😅

```suggestion
    ..addOption(
      _directoryFlag,
      defaultsTo: 'data',
      abbr: 'd',
      help: 'Directory to download data into',
    )
```